spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -121,6 +121,15 @@ public Table loadTable(Identifier ident) throws NoSuchTableException {
     }
   }
 
+  @Override
+  public void invalidateTable(Identifier ident) {
+    if (icebergCatalog.tableExists(ident)) {
+      icebergCatalog.invalidateTable(ident);
+    } else {
+      getSessionCatalog().invalidateTable(ident);
+    }
```

Comment:
Nit / non-blocking: Is there any need to check if the `Identifier` belongs to the catalog of `icebergCatalog`? I doubt it because tableExists should be checking that, but if there's no test around that might consider adding one (or just trying it out).

Reply:
It's basically the same as the "loadTable" method in this class. TableExists just catches the noSuchTable exception and returns false

https://github.com/apache/iceberg/blob/b623b074c6242ac3b48fce8712dedd7f97e6ee5b/api/src/main/java/org/apache/iceberg/catalog/Catalog.java#L268-L275

So I don't really think we need a check here. Again it's basically the same logic for "loadTable" so if loadTable would touch this table in the Iceberg catalog, refresh table will also use the Iceberg table.